src/plugins/controls/public/options_list/embeddable/options_list_embeddable.tsx

```diff
@@ -322,6 +322,12 @@ export class OptionsListEmbeddable extends Embeddable<OptionsListEmbeddableInput
           },
           this.abortController.signal
         );
+      if (rejected) {
+        // This prevents a rejected request (which can happen, for example, when a user types a search string too quickly)
+        // from prematurely setting loading to `false` and updating the suggestions to show "No results"
+        return;
```

Comment:
To test this, remove this early return, throttle your network, and type something quickly into the search bar - you'll notice that you'll see a flash of the loading state, then you'll see the `"No results"` message which will hang, before finally seeing your expected results like so:

![Jan-05-2023 10-10-13](https://user-images.githubusercontent.com/8698078/210839785-4086ce16-04b1-4cd5-90b8-83e5619a64f7.gif)

With this early return, the earlier rejected requests do not set `loading` to `false` until results are actually received. So, you get expected the following more expected behaviour:

![Jan-05-2023 10-13-34](https://user-images.githubusercontent.com/8698078/210840423-b219bdd6-f5c0-4e47-bfc3-a10a8a1e8d0c.gif)

From what I can tell, it seems like the requests are rejected due to the `debounce` on search string changes... Seems like we make a request for every character typed, but cancel previous requests if they haven't yet returned results in favour of the newer requests. Not **entirely** sure that this is what is happening, but it's my best guess after some deep debugging 👀 

It might be possible to instead prevent these rejected results **entirely** by debouncing the **request** rather than the search string changes.... Figured I'd go with the easier solution (this early return) for now 🤷